File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlAliasExpr.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression } from './ISqlExpression';

export class SqlAliasExpr implements ISqlExpression {
    SqlType: SqlType = SqlType.AliasExpr;
    Span: TextSpan = new TextSpan();
    Name: string = '';

    Accept(visitor: any): void {
        // 簡單實作
    }

    ToSql(): string {
        return `AS ${this.Name}`;
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlAliasExpr.cs
namespace T1.SqlSharp.Expressions;

public class SqlAliasExpr : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.AliasExpr;
    public TextSpan Span { get; set; } = new();
    public required string Name { get; set; } = string.Empty;
    public string ToSql()
    {
        return $"AS {Name}";
    }
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_AliasExpr(this);
    }
}